### DISCUSSION TITLE
why does this need a browser

### DISCUSSION THREAD

**Discussion by sewnie** [charmbracelet/vhs#291] [General]
i expect a program like vhs, 'CLI home video recorder', to record from the .tape file and then output to a gif.

why the shit does it require a browser?

**Comment by maaslalani** [comment]
Hey @apprehensions, VHS uses the browser to perform the actions of the tape file and to run the commands. The commands that are typed and run are actually executed in a terminal. Therefore, we would need a terminal to execute the commands in. For this, we use `ttyd` and run the commands in the browser as well to record the terminal to capture the GIF frames that way you can also customize the terminal to make the GIF match the appearance you want.

**Reply by sewnie** [reply]
i assume this is to remove the cost of having to maintain an terminal emulator recorder, instead of using a adequate terminal emulator, and just capturing its output, similar to `st -o`

**Reply by maaslalani** [reply]
Yep, and also to allow it to work across multiple platforms like MacOS and Windows rather than just linux.

**Reply by maaslalani** [reply]
I do think there is room for a version that does not use the browser to consume less memory / compute so that VHS can be run on less powerful machines and have better frame rate capture.
